pkg/apis/rollouts/v1alpha1/analysis_types.go

```diff
@@ -47,11 +47,11 @@ type AnalysisMetric struct {
 	// If both success and failure conditions are specified, and the measurement does not fall into
 	// either condition, the measurement is considered Inconclusive
 	FailureCondition string `json:"failureCondition,omitempty"`
+	// Count is the number of times to run measurement. If omitted, runs indefinitely
+	Count int32 `json:"count,omitempty"`
 	// MaxFailures is the maximum number of times the measurement is allowed to fail, before the
-	// entire metric is considered failed (default: 1)
-	MaxFailures *int32 `json:"maxFailures,omitempty"`
-	// FailFast will fail the entire analysis run prematurely
-	FailFast bool `json:"failFast,omitempty"`
```

Comment:
FailFast was removed because any failed metric should fail the run as soon as possible.